numpy/lib/tests/test_arraypad.py

```diff
@@ -1122,8 +1155,75 @@ def _padwithtens(vector, pad_width, iaxis, kwargs):
         assert_array_equal(a, b)
 
 
-class TestNdarrayPadWidth(object):
-    def test_check_simple(self):
+class TestUnicodeInput(object):
+    def test_unicode_mode(self):
+        constant_mode = u'constant'
+        a = np.pad([1], 2, mode=constant_mode)
+        b = np.array([0, 0, 1, 0, 0])
+        assert_array_equal(a, b)
+
+
+@pytest.mark.parametrize("mode", ["edge", "symmetric", "reflect", "wrap"])
+def test_object_input(mode):
+    # Regression test for issue gh-11395.
+    a = np.full((4, 3), fill_value=None)
+    pad_amt = ((2, 3), (3, 2))
+    b = np.full((9, 8), fill_value=None)
+    assert_array_equal(pad(a, pad_amt, mode=mode), b)
+
+
+class TestPadWidth(object):
+
+    @pytest.mark.parametrize("pad_width", [
+        (4, 5, 6, 7),
+        ((1,), (2,), (3,)),
+        ((1, 2), (3, 4), (5, 6)),
+        ((3, 4, 5), (0, 1, 2)),
+    ])
+    @pytest.mark.parametrize("mode", _all_modes.keys())
```

Comment:
Using `parametrize` instead of for loops makes the code more readable but really bloats the number of unit tests and thus the output of pytest. What's the general preference here?

Reply:
I think we tend to use parameterize now. If it really does bloat output too much, maybe we can also ask pytest to add some nobs to help.

Reply:
Okay, good to know.